modules/tinymce/src/themes/silver/test/ts/browser/view/ViewTest.ts

```diff
@@ -0,0 +1,337 @@
+        });
+
+        editor.ui.registry.addContextToolbar('test-context', {
+          predicate: (node) => node.nodeName.toLowerCase() === 'img',
+          items: 'bold'
+        });
+      }
+    }, []);
+
+    const clickViewButton = (editor: Editor, tooltip: string) => {
+      const btn = UiFinder.findIn(TinyDom.container(editor), `.tox-view button[title='${tooltip}']`).getOrDie();
+      Mouse.click(btn);
+    };
+
+    const toggleView = (name: string) => {
+      const editor = hook.editor();
+      editor.execCommand('ToggleView', false, name);
+    };
+
+    const queryToggleView = () => {
+      const editor = hook.editor();
+      return editor.queryCommandValue('ToggleView');
+    };
+
+    const assertMainViewHidden = () => {
+      const editor = hook.editor();
+      const editorContainer = UiFinder.findIn(TinyDom.container(editor), '.tox-editor-container').getOrDie();
+
+      assert.equal('true', Attribute.get(editorContainer, 'aria-hidden'), 'Should be aria-hidden');
+      assert.equal('none', Css.getRaw(editorContainer, 'display').getOrDie(), 'Should have display none');
+    };
+
+    const assertMainViewVisible = () => {
+      const editor = hook.editor();
+      const editorContainer = UiFinder.findIn(TinyDom.container(editor), '.tox-editor-container').getOrDie();
+
+      assert.isFalse(Attribute.has(editorContainer, 'aria-hidden'), 'Should not have aria-hidden');
+      assert.isTrue(Css.getRaw(editorContainer, 'display').isNone(), 'Should not have display none');
+    };
+
+    const assertViewHtml = (viewIndex: number, expectedHtml: string) => {
+      const editor = hook.editor();
+      const editorContainer = UiFinder.findIn<HTMLElement>(TinyDom.container(editor), `.tox-view:nth-child(${viewIndex + 1}) .tox-view__pane`).getOrDie();
+
+      assert.equal(Html.get(editorContainer), expectedHtml);
+    };
+
+    const pWaitUntilRemoved = (label: string, selector: string) =>
+      Waiter.pTryUntil(label, () => UiFinder.notExists(SugarBody.body(), selector));
+
+    const pAssertToolbarDrawerVisibleState = async (expectedState: boolean) => {
+      if (expectedState) {
+        await UiFinder.pWaitForVisible('Wait for toolbar drawer to be visible', SugarBody.body(), '.tox-toolbar__overflow');
+      } else {
+        await pWaitUntilRemoved('Wait for toolbar drawer to close', '.tox-toolbar__overflow');
+      }
+    };
+
+    it('TINY-9210: Structure', () => {
+      const editor = hook.editor();
+      const viewWrap = UiFinder.findIn(TinyDom.container(editor), '.tox-view-wrap').getOrDie();
+
+      Assertions.assertStructure('Checking structure', ApproxStructure.build((s, str, arr) => {
+        const button = (title: string, classes: string[]) =>
+          s.element('button', {
+            classes: Arr.map(classes, (cls) => arr.has(cls)),
+            attrs: { 'title': str.is(title), 'type': str.is('button'), 'tabindex': str.is('-1'), 'data-alloy-tabstop': str.is('true') },
```

Comment:
This would probably be easier to read split onto separate lines (the attrs object)